### PR TITLE
Add cNFT support to Explorer

### PR DESCRIPTION
**Pull request by ngundotra**
- [x] account header

Screenshot 2024-05-24 at 12 15 31 PM

- [x] account data
- [x] generic cache
- [x] cnft tabs (metadata, attributes)

- [x] additional compression tab to show relevant proof data

Screenshot 2024-05-24 at 12 11 25 PM

Screenshot 2024-05-24 at 12 07 49 PM

Examples:
- https://explorer.solana.com/address/4MupsdStMNTVwYzGUq75BkPMNGFLdircExr1aGT7eAiG
- https://explorer.solana.com/address/2qGxZpFqCKtgcXNhMJtxAv5CRMHzUEvzHB6ae3bK4MZq

### CODE REVIEW

File: app/components/account/CompressedNftCard.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+                    <td className="text-lg-end">
+                        <a rel="noopener noreferrer" target="_blank" href={compressedNft.content.links.external_url}>
+                            {compressedNft.content.links.external_url}
+                            <ExternalLink className="align-text-top ms-2" size={13} />
+                        </a>
+                    </td>
+                </tr>
+                <tr>
+                    <td>Seller Fee</td>
+                    <td className="text-lg-end">{`${compressedNft.royalty.basis_points / 100}%`}</td>
```
Comment (jnwng) [inline app/components/account/CompressedNftCard.tsx:65]: round this to two decimal points or you get weird overflows

File: app/components/account/CompressedNftCard.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+            }
+            return new Dropdown(dropdownRef.current);
+        },
+        dropdown => {
+            if (dropdown) {
+                dropdown.dispose();
+            }
+        },
+        [dropdownRef]
+    );
```
Comment (jnwng) [inline app/components/account/CompressedNftCard.tsx:108]: this code smells, there's really no other dropdown code?

File: app/components/account/CompressedNftCard.tsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+                    <div className="dropdown-menu mt-2">{getCreatorDropdownItems(compressedNft.creators)}</div>
+                </div>
+            </div>
+        </div>
+    );
+}
+
+function getCompressedNftPill() {
+    const onchainVerifiedToolTip =
+        'This NFT does not have on-chain data, but can be transferred and traded on-chain normally. This tag guarantees that this compressed NFT has the correct format.';
```
Comment (jnwng) [inline app/components/account/CompressedNftCard.tsx]: this is slightly confusing, here's a suggestion:
```
This NFT does not have a corresponding account, but uses verified ledger data to allow for transfers and trades. The existence of this tag ensures that the compressed NFT is verifiably up-to-date with the chain.
```

File: app/providers/compressed-nft.tsx [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    async (url: string) => {
+        return fetch(url).then(response => response.json());
+    }
+);
+
+export const useCompressedNft = makeCache<{ address: string; url: string }, CompressedNft | null>(
+    'compressedNft',
+    ({ address, url }) => `${address}-${url}`,
+    async ({ address, url }) => {
+        return fetch(`${url}`, {
```
Comment (jnwng) [inline app/providers/compressed-nft.tsx:58]: neither here nor there but https://www.npmjs.com/package/@tanstack/react-query or [useSWR](https://blog.logrocket.com/handling-data-fetching-next-js-useswr/) might be better abstractions for caching these API calls

File: app/providers/compressed-nft.tsx [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+          jsonrpc: string;
+          id: string;
+          error: {
+              code: number;
+              message: string;
+          };
+      };
+
+export type CompressedNft = {
+    interface: string;
```
Comment (jnwng) [inline app/providers/compressed-nft.tsx:123]: any chance we can pull this from one of the types declarations, perhaps from https://www.npmjs.com/package/@metaplex-foundation/digital-asset-standard-api

**Review by jnwng — CHANGES_REQUESTED** [review]

File: app/providers/compressed-nft.tsx (hunk 4 above)
Comment (mcintyre94) [reply app/providers/compressed-nft.tsx:58]: We do use SWR in some other places FWIW

File: app/components/account/CompressedNftCard.tsx (hunk 2 above)
Comment (ngundotra) [reply app/components/account/CompressedNftCard.tsx:108]: reusing existing dropdown code for normal creators dropdown for metadata

File: app/providers/compressed-nft.tsx (hunk 5 above)
Comment (ngundotra) [reply app/providers/compressed-nft.tsx:123]: Adding modern deps is PITA. Worth doing if we also upgrade anchor (using `project-serum/anchor` still)

File: app/components/account/CompressedNftCard.tsx (hunk 1 above)
Comment (ngundotra) [reply app/components/account/CompressedNftCard.tsx:65]: Should be alright bc `basis_points` never has decimals

File: app/components/account/CompressedNftCard.tsx (hunk 3 above)
Comment (ngundotra) [reply app/components/account/CompressedNftCard.tsx]: Adding

**Pull request merged**